tests/shared/utils/test_io.py

```diff
@@ -343,6 +358,17 @@ def test_create_directory_for_file(tmp_path: Path):
     assert os.path.exists(os.path.dirname(file))
 
 
+def test_write_utf_8_yaml_file(tmp_path: Path):
+    """This test makes sure that dumping a yaml doesn't result in Uxxxx sequences
+    but rather directly dumps the unicode character."""
+
+    file_path = str(tmp_path / "test.yml")
```

Comment:
do we even need the `str` cast?

Reply:
I just moved that code - don't think so, will remove it

Reply:
Left it as is - thinking maybe this was tested with str on purpose (it is in shared)